sfm_pc/management/commands/import_google_doc.py

```diff
@@ -49,6 +49,57 @@
 from location.models import Location
 
 
+class EntityMap(dict):
+    '''
+    Container for mapping of UUID to array of (name, column, row, sheet) tuples.
+    '''
+    KEY_TYPE = 'UUID'
+    VALUE_TYPE = 'name'
+
+    def add(self, key, value, column, row, sheet):
```

Comment:
Seems like this might be a bit easier to reason about if you implemented this as a nested dict

```python
__init__(self):
    _d = {}
...
if key not in self:
    self._d[key] = {}

location = (column, row, sheet)
if value not in self[key]:
    self_d[key][value] = [location]
else:
    self._d[key][value].append(location)

return self
```

if you need the data back in a particular way you are getting it now you could
implement this dunder method

```python
def __getitem__(self, key):
    return [(k,) + values for k, values in self._d[key].items()]
```

Reply:
Hmm, I'm finding this suggestion much harder to parse. I originally implemented the map as an object with a dictionary attribute here, https://github.com/security-force-monitor/sfm-cms/pull/777/commits/e182853297003718d76ecd33d6373852803ec89b, but I found that most times I accessed `self`, I was accessing `self.map` (the dictionary), so it felt more direct for `EntityMap` to be a dictionary itself. Is this a change you require?

Reply:
i think you effectively do have a nested structure since the logic, as, I understand it is to have

```
key
|_ value
    |_ location
```

which you sometimes want to transpose to another nested structure:

```
value
|_ key
    |_ location
```


i think it's going to be easier to work with this in the future if your data structure reflects that nesting.

so, i do want to see an implementation that has that nesting.

I do not require that you implement this wit had dictionary attribute. that's only necessary if you need to flattened results back as in the example ```__getitem__``` if you don't need that flattening, you could do.

```
if key not in self:
    self[key] = {}

location = (column, row, sheet)
if value not in self[key]:
    self[key][value] = [location]
else:
    self[key][value].append(location)

return self
```

Reply:
Thanks for clarifying! I missed the substance of your first comment. 😅 Refactored in https://github.com/security-force-monitor/sfm-cms/pull/777/commits/ad9d10b5a749d85340ac5c6972fd14031b480c5e.